### ISSUE TITLE
bug: old agents can persist across `resize-world`

### ISSUE THREAD

**Issue by SethTisue** [NetLogo/NetLogo#389]
example failing test cases:

```
ResizeWorldHuntDownDeadPatches
  O> set glob1 patch max-pxcor max-pycor
  O> resize-world 0 0 0 0
  glob1 => nobody

ResizeWorldSpecialAgentSets
  O> crt 10
  O> set glob1 turtles
  O> resize-world 0 0 0 0
  count turtles => 0
  count glob1 => 0
```

how to fix: maybe we could call `realloc` and `realloc` could walk through all of the global variables and find any patches and replace them with nobody and find any patchsets and replace them with the empty patchset (to really get absolutely everything we'd need to walk all local variables and let bindings in all stack frames, too, and look inside closed-over values in tasks. perhaps excessive, but otoh, if we had infrastructure for doing that it might be useful for other things)

**Comment by nicolaspayette** [comment]
Testing this in 6.0, it seems like the problem still occurs with patches:

```
observer> set glob1 patch max-pxcor max-pycor
observer> resize-world 0 0 0 0
observer> show glob1
observer: (patch 16 16)
```
but not with turtles:
```
observer> crt 10
observer> set glob1 turtles
observer> resize-world 0 0 0 0
observer> show count turtles
observer: 0
observer> show count glob1
observer: 0
```

**Comment by alan-isaac** [comment]
Turtles persist if resize-world does not change the world dimensions.  (This is either a bug or a documentation bug.)

**Comment by LaCuneta** [comment]
@alan-isaac Good point, I've added some language to the `resize-world` documentation to indicate that turtles will live if the world size doesn't change.

**Comment by ERSUCC** [comment]
I was able to reproduce the behavior most recently described by @nicolaspayette in NetLogo 7. Turns out the problem was that old patches were not being marked as dead after resizing the world. Doing so fixed the problem in all cases mentioned above without the need for stack traversal, because dead agents can't be accessed by any primitives.

**Comment by SethTisue** [comment]
heh, I don't remember the concept of a "dead patch" ever arising before! but it does seem to make sense here

**Comment by ERSUCC** [comment]
It's not logically the best solution, but the infrastructure for it is there already and it aligns with the behavior of other agents in these sorts of situations, so it ended up being the simplest solution.
